### PR TITLE
Fix sjoin with duplicate fieldnames

### PR DESCRIPTION
**Pull request by brahn**
See https://github.com/chrispahm/spatialmerge/issues/2

### CODE REVIEW

**Review by chrispahm — APPROVED** [review]

File: test/test.js [hunk 1]
```diff
@@ -32,7 +33,7 @@ const readFile = util.promisify(fs.readFile)
   // expected output data
   const mergeExpected = require('./output/merged.json')
   const citiesExpected = require('./output/merged_cities.json')
-  const plotsExpected = await reader('./test/output/merged_plots.json')
+  // const plotsExpected = await reader('./test/output/merged_plots.json')
```
Comment (chrispahm) [inline test/test.js]: Yes, this was an old test file, look like I forgot to delete the line accordingly

File: src/helpers.js [hunk 2]
```diff
@@ -21,13 +36,14 @@ function mergeWith (object1, object2, lsuffix, rsuffix) {
       if (duplicates.indexOf(key) > -1) {
         feature[key + '_' + rsuffix] = object2[key]
       } else {
-        feature[key] = object1[key]
+        feature[key] = object2[key]
```
Comment (chrispahm) [inline src/helpers.js]: That's right, this has been a copy/paste bug

**Review by chrispahm — COMMENTED** [review]
LGTM! However, we should add a test using the data from #2 to ensure that sjoin returns the same output as geopandas. I'll work on that!

File: src/helpers.js [hunk 3]
```diff
@@ -14,29 +14,29 @@ function findAllDuplicates(array) {
   return [...duplicates];
 }
 
-function mergeWith(object1, object2, lsuffix = 'left', rsuffix = 'right') {
-  const keys1 = Object.keys(object1)
-  const keys2 = Object.keys(object2)
+function mergeWith(leftFeatureProps, rightFeatureProps, lsuffix = 'left', rsuffix = 'right') {  
+  const keys1 = Object.keys(leftFeatureProps)
+  const keys2 = Object.keys(rightFeatureProps)
```
Comment (brahn) [inline src/helpers.js:19]: good call!

File: test/test.js [hunk 4]
```diff
@@ -8,7 +8,11 @@ import { strictEqual, deepStrictEqual } from 'assert'
 import reader from './utilities/reader.js'
 const require = createRequire(import.meta.url)
 const readFile = util.promisify(fs.readFile)
-// const writeFile = util.promisify(fs.writeFile)
+const writeFile = util.promisify(fs.writeFile)
+
+const sortBy = (fn) => {
+  return (a, b) => (fn(a) > fn(b)) ? 1 : ((fn(b) > fn(a)) ? -1 : 0);
+}
```
Comment (brahn) [inline test/test.js:15]: I tend to pull in https://underscorejs.org/ for this and other niceties but this works just fine :-)

**Review by brahn — COMMENTED** [review]
Looks good to me!

**Pull request merged**